src/python/pim/polynomial_representation.py

```diff
@@ -28,12 +28,12 @@ def __mul__(self, other: "Polynomial") -> "Polynomial":
         (convention that a zero polynomial has a degree of 0)
         """
         if Polynomial.find_degree(self) == -1:
-            degree_self: int = 0
+            degree_self: int = 1
```

Comment:
I take it this is the change that fixed the `IndexError`? I presume all the tests pass now.